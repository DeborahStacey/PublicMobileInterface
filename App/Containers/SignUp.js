import React from 'react'
import { ScrollView, Text, Image, View, TextInput } from 'react-native'
import { Images } from '../Themes'
import RoundedButton from '../Components/RoundedButton'
// import { Actions as NavigationActions } from 'react-native-router-flux'
import {create} from 'apisauce'

// Styles
import styles from './Styles/PresentationScreenStyle'

const db = create({
  baseURL: 'http://cat.ddns.net/Backend/api.php',
  headers: {'Content-Type': 'application/json'}
})

export default class SignUp extends React.Component {

  constructor (props) {
    super(props)
    this.state = {
      fName: '',
      lName: '',
      email: '',
      password: '',
      address: ''
    }
  }

  registerUser () {
    if (this.state.email === '' || this.state.password === '' || this.state.fName === '' || this.state.lName === '' || this.state.address === '') {
      window.alert('not all of the fields have been filled out please enter all data!')
      return
    }

    var postObj = {
      'email': this.state.email,
      'password': this.state.password,
      'firstName': this.state.fName,
      'lastName': this.state.lName,
      'address': {
        'street': this.state.address,
        'unit': ' ',
        'city': ' ',
        'postalCode': ' ',
        'locationID': 13
      }
    }

    db.post('/user/register', postObj)
    .then((response) => window.alert(JSON.stringify(response.data)))
  }

  updateFName (event) {
    this.setState({
      fName: event
    })
  }

  updateLName (event) {
    this.setState({
      lName: event
    })
  }

  updateEmail (event) {
    this.setState({
      email: event
    })
  }

  updatePassword (event) {
    this.setState({
      password: event
    })
  }

  updateAddress (event) {
    this.setState({
      address: event
    })
  }
  
  focusNextField = (nextField) => {
    this.refs[nextField].focus()
  }

  render () {
    return (
      <View style={styles.mainContainer}>
        <Image source={Images.background} style={styles.backgroundImage} resizeMode='stretch' />
        <ScrollView style={styles.container}>

          <View style={styles.messageBox}>
            <View style={styles.messageBoxContents}>

              <Text style={styles.sectionText}>First Name</Text>
              <TextInput ref="1" onSubmitEditing={() => this.focusNextField('2')} returnKeyType="next" onChangeText={this.updateFName.bind(this)} placeholder='Enter your first name' placeholderTextColor='white' style={styles.sectionInput} />

              <Text style={styles.sectionText}>Last Name</Text>
              <TextInput ref="2" onSubmitEditing={() => this.focusNextField('3')} returnKeyType="next" onChangeText={this.updateLName.bind(this)} placeholder='Enter your last name' placeholderTextColor='white' style={styles.sectionInput} />

              <Text style={styles.sectionText}>Email</Text>
              <TextInput ref="3" onSubmitEditing={() => this.focusNextField('4')} returnKeyType="next" onChangeText={this.updateEmail.bind(this)} placeholder='Enter your email' placeholderTextColor='white' style={styles.sectionInput} keyboardType="email-address" />

              <Text style={styles.sectionText}>Password</Text>
              <TextInput ref="4" onSubmitEditing={() => this.focusNextField('5')} returnKeyType="next" onChangeText={this.updatePassword.bind(this)} placeholder='Enter your password' placeholderTextColor='white' style={styles.sectionInput} secureTextEntry={true} />

              <Text style={styles.sectionText}>Address</Text>
              <TextInput ref="5" onSubmitEditing={this.registerUser.bind(this)} returnKeyType="done" onChangeText={this.updateAddress.bind(this)} placeholder='Enter your address' placeholderTextColor='white' style={styles.sectionInput}  />

            </View>
          </View>

          <RoundedButton onPress={this.registerUser.bind(this)}>
            Register
          </RoundedButton>

        </ScrollView>
      </View>
    )
  }
}
