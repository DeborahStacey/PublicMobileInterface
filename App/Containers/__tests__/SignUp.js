import 'react-native'
import React from 'react'
import SignUp from '../SignUp'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <SignUp />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
