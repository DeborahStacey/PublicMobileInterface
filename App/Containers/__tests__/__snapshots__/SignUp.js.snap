exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F5FF",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <Image
    resizeMode="stretch"
    source={undefined}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    } />
  <ScrollView
    style={
      Object {
        "flex": 1,
        "paddingTop": 10,
      }
    }>
    <View
      style={undefined}>
      <View
        style={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2C457F",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }>
          First Name
        </Text>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Enter your first name"
          placeholderTextColor="white"
          returnKeyType="next"
          style={
            Object {
              "color": "#2C457F",
              "fontSize": 14,
              "fontWeight": "bold",
              "height": 50,
              "marginVertical": 5,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2C457F",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }>
          Last Name
        </Text>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Enter your last name"
          placeholderTextColor="white"
          returnKeyType="next"
          style={
            Object {
              "color": "#2C457F",
              "fontSize": 14,
              "fontWeight": "bold",
              "height": 50,
              "marginVertical": 5,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2C457F",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }>
          Email
        </Text>
        <TextInput
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Enter your email"
          placeholderTextColor="white"
          returnKeyType="next"
          style={
            Object {
              "color": "#2C457F",
              "fontSize": 14,
              "fontWeight": "bold",
              "height": 50,
              "marginVertical": 5,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2C457F",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }>
          Password
        </Text>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Enter your password"
          placeholderTextColor="white"
          returnKeyType="next"
          secureTextEntry={true}
          style={
            Object {
              "color": "#2C457F",
              "fontSize": 14,
              "fontWeight": "bold",
              "height": 50,
              "marginVertical": 5,
            }
          } />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2C457F",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }>
          Address
        </Text>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Enter your address"
          placeholderTextColor="white"
          returnKeyType="done"
          style={
            Object {
              "color": "#2C457F",
              "fontSize": 14,
              "fontWeight": "bold",
              "height": 50,
              "marginVertical": 5,
            }
          } />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#466ECC",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 25,
          "marginVertical": 10,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginVertical": 10,
            "textAlign": "center",
          }
        }>
        REGISTER
      </Text>
    </View>
  </ScrollView>
</View>
`;
